hr_timesheet_overview/models/hr_employee_hour.py

```diff
@@ -0,0 +1,327 @@
+            "date": timesheet.date,
+            "employee_id": timesheet.employee_id.id,
+            "project_id": project.id,
+            "analytic_group_id": timesheet.group_id.id,
+            "hours_qty": hours_qty,
+            "days_qty": days_qty,
+            "sum_type": "add",
+        }
+        return values
+
+    @api.model
+    def _prepare_timesheets_lines(self, employee, date_from):
+        """Retrieve all timesheets values
+
+        :param employee: an employee record
+        :param date_from: a datetime.date object (default first contract date, included)
+        """
+        aal_model = self.env["account.analytic.line"]
+        timesheets = aal_model.search(
+            [
+                ("employee_id", "=", employee.id),
+                ("project_id", "!=", False),
+                ("unit_amount", "!=", 0.0),
+                ("date", ">=", date_from),
+            ]
+        )
+        values_list = []
+        for timesheet in timesheets:
+            values = self._prepare_timesheet_line(timesheet)
+            if values:
+                values_list.append(values)
+        return values_list
+
+    @api.model
+    def action_generate_data(self, employee_ids=None):
+        """Launch update process
+        In case of force update, we unlink all related lines and reprocess them
+        completely
+
+        :param employee_ids: a list of employee ids
+        :param date_from: a datetime.date object (default first contract date, included)
+        :param date_to: a datetime.date object (default today, included)
+        :param force: purge all lines related to params before regenerating
+        """
+        employees = self.env["hr.employee"].browse(employee_ids)
+        employees = employees.filtered(lambda emp: emp.last_hours_report_date)
+        if not employees:
+            employees = self.env["hr.employee"].search(
+                [
+                    ("last_hours_report_date", "!=", False),
+                ]
+            )
+
+        for employee in employees:
+            date_from = employee.last_hours_report_date
+            existing_lines = self.search(
+                [
+                    ("employee_id", "=", employee.id),
+                    ("date", ">=", date_from),
+                ]
+            )
+            if existing_lines:
+                _logger.info(f"Purge hours for '{employee.name}' from {date_from}")
+                existing_lines.unlink()
+            _logger.info(f"Generating hours for '{employee.name}' from {date_from}")
+            self._create_values_per_employee(employee, date_from)
+            employee.contract_ids.update(
+                {"last_hours_report_date": fields.Date.today()}
+            )
+
+    def _create_values_per_employee(self, employee, date):
+        att_vals = self._get_attendances_values(employee, date)
+        ts_vals = self._prepare_timesheets_lines(employee, date)
+        self.create(att_vals + ts_vals)
```

Comment:
Be carefull of the possibility to extend this method by adding more records of different models as the original module is made for.
```suggestion

    @api.model
    def _prepare_values(self, employee):
        """ Return a list of dict with computed values for this employee

        :param employee: an employee record
        """
        attendances_by_date = self.env.context.get("attendances_by_date", {})
        timesheet_values = self._prepare_timesheet_lines(employee)
        return list(attendances_by_date.values()) + timesheet_values
        
    def _create_values_per_employee(self, employee, date):
        attendances_by_date = self._get_attendances_values_by_date(employee)
        values = self.with_context(
            attendances_by_date=attendances_by_date,
            active_test=False  # We want every records, even inactive
        )._prepare_values(employee)
        self.create(values)
```

Reply:
I prefer to keep my version